test/tts/tts.hpp

```diff
@@ -432,6 +432,19 @@ namespace tts
   };
   template<typename... Ls> struct concatenate { using type = decltype( (Ls{} + ...) ); };
   template<typename... Ls> using concatenate_t = typename concatenate<Ls...>::type;
+  // filter types with predicate
+  template<template<typename> typename Pred, typename Type> struct filter
+  {
+    template<typename... Ls> static constexpr types<Ls...> tuple_to_types(const std::tuple<Ls...>&);
+    template<typename T>
+    static constexpr std::conditional_t<Pred<T>::value, std::tuple<T>, std::tuple<>> filter_type();
+
+    template<typename... Ls> static constexpr auto filter_impl(types<Ls...>)
+    {
+      return tuple_to_types(std::tuple_cat(filter_type<Ls>()...));
+    }
+    using type = decltype(filter_impl(Type {}));
+  };
```

Comment:
I appreciate you doing this, shame it wasn't there.
Should be done upstream

@jfalcou - can you implement filter? or what do you think?

Reply:
I can it to main tts yes

Reply:
then we let this implementation be here for now